lib/ansible/template/__init__.py

```diff
@@ -231,6 +232,30 @@ def recursive_check_defined(item):
             raise AnsibleFilterError("{0} is undefined".format(item))
 
 
+def _unroll_generator(func):
+    """Wrapper function, that intercepts the result of a filter
+    and auto unrolls a generator, so that users are not required to
+    explicitly use ``|list`` to unroll.
+    """
+    def wrapper(*args, **kwargs):
+        ret = func(*args, **kwargs)
+        if USE_JINJA2_NATIVE and isinstance(ret, types.GeneratorType):
+            return list(ret)
+        return ret
+
+    for attr in ('__module__', '__name__', '__qualname__', '__doc__', '__annotations__'):
```

Comment:
Need to add a comment about why I'm not just using `functools.wraps` or `functools.update_wrapper`. This is a copy from py3.7, that doesn't traceback.